docs/introduction-to-smart-contracts.rst

```diff
@@ -51,7 +51,7 @@ code that manages the database. In this example, the contract defines the
 functions ``set`` and ``get`` that can be used to modify
 or retrieve the value of the variable.
 
-To access a state variable, you do not need the prefix ``this.`` as is common in
+To access a state variable, you can not prefix with ``this.`` as is common in
 other languages.
```

Comment:
How about this?
```markdown
State variables can be accessed without the ``this.`` prefix.
Note that omitting it is not just a matter of style.
Calling a function or accessing a variable through a contract instance results in an external call,
which is more expensive and cannot be used to modify the variable.
```

Reply:
I like the first half as it says the usage of `this` is unlike other languages and omitting it is not just a matter of style. The second half is too esoteric.

How about this?
```
State variables can be accessed without the ``this.`` prefix.
Note that, unlike in some other languages, omitting it is not just a matter of style.
You can actually use `this` but only for public variables (for external you need `this`). 
For internal, you can't use `this`.
```

Reply:
Well, if you think that going into details in unnecessary then we could just change it to this:
```rst
To access a state variable, you do not typically add the ``this.`` prefix.
Unlike in some other languages, omitting it is not just a matter of style,
it results in a completely different operation.
```

But if you want it to say why, here's another version:
```rst
To access a state variable, you do not typically add the ``this.`` prefix.
Unlike in some other languages, omitting it is not just a matter of style.
Using ``this`` you can only access its :ref:`getter function <getter-functions>`
that is only provided for public state variables.
```

Note that `` :ref:`getter function <getter-functions>` `` is RST syntax for a link to [Contracts > Getter Functions](https://docs.soliditylang.org/en/latest/contracts.html#getter-functions).

Reply:
```... it results in a completely different operation. ``` is good for me!